pkg/operation/botanist/operatingsystemconfig.go

```diff
@@ -191,11 +168,133 @@ func (b *Botanist) computeOperatingSystemConfigsForWorker(machineTypes []gardenv
 		"secretName":           secretName,
 		"customization":        customization,
 	}
+	var (
+		evictionHard            = map[string]string{}
+		evictionSoft            = map[string]string{}
+		evictionSoftGracePeriod = map[string]string{}
+		evictionMinimumReclaim  = map[string]string{}
+	)
+
+	// use the spec.Kubernetes.Kubelet as default for worker
+	kubeletConfig := worker.Kubelet
+	if kubeletConfig == nil {
+		kubeletConfig = b.Shoot.Info.Spec.Kubernetes.Kubelet
+	}
+
+	// ensure sane defaults for evictionHard.memoryAvailable and evictionSoft.memoryAvailable
+	evictionHard["memoryAvailable"], evictionSoft["memoryAvailable"] = getEvictionMemoryAvailable(machineTypes, worker.MachineType)
+
+	if kubeletConfig := kubeletConfig; kubeletConfig != nil && kubeletConfig.EvictionHard != nil {
```

Comment:
Why always checking if `kubeletConfig` is not `nil`?
Why not (modulo variable names improvement mentioned before):

```golang
if kubeletConfig := kubeletConfig; kubeletConfig != nil {
  if kubeletConfig.EvictionHard != nil {
    ...
  }

  if kubeletConfig.EvictionSoft != nil {
    ...
  }

  ...
}
```